Components/webComponents/cartComponents/freightOptionsComponent.js

```diff
@@ -18,58 +23,54 @@ export class FreightOptions extends HTMLElement {
+        if (name === "selectedfreightoption") {
+            this.selectedFreightOption = newValue;
+            this.render();
+        }
     }
 
-    renderFreightOptions(freightOptions, cartFreightResponse) {
+    render() {
         let renderContainer = this.querySelector("#cart-freight-options");
         renderContainer.innerHTML = "";
-        for (var i = 0; i < freightOptions.length; i++) {
-            let freightOption = freightOptions[i];
-            let freightName;
-            let freightInput = document.createElement("input");
+        for (let i = 0; i < this.freightOptions.length; i++) {
+            const freightOption = this.freightOptions[i];
+            const freightName = this.freightNames[freightOption.name];
+            const freightInput = document.createElement("input");
             freightInput.setAttribute("name", "radio-input");
             freightInput.setAttribute("class", "radio-input");
             freightInput.setAttribute("type", "radio");
             freightInput.setAttribute("id", freightOption.name);
             freightInput.setAttribute("value", freightOption.name);
-            freightInput.checked =
-                freightOption.name ===
-                cartFreightResponse.selectedFreightOption;
+            freightInput.checked = freightOption.name === this.selectedFreightOption;
 
-            let listItem = document.createElement("li");
-            let freightLabel = document.createElement("label");
+            const listItem = document.createElement("li");
+            const freightLabel = document.createElement("label");
             freightLabel.setAttribute("class", "radio-label");
             freightLabel.setAttribute("for", freightOption.name);
-
-            if (freightOption.name === "PickupInStore") {
-                freightName = "Pick Up in Store";
-            } else if (freightOption.name === "Porterbuddy") {
-                freightName = "Porterbuddy";
-            } else if (freightOption.name === "PickupPoint") {
-                freightName = "Pickup Point Oslo / Sandefjord";
-            } else if (freightOption.name === "Letter") {
-                freightName = "Letter";
-            }
-
+            
             freightLabel.innerHTML = `${freightName}: ${freightOption.price},-`;
             renderContainer.appendChild(listItem);
             listItem.appendChild(freightInput);
             listItem.appendChild(freightLabel);
 
             freightInput.addEventListener("click", async function () {
-                let cartApi = new CartApi();
-                await cartApi.freightOptionSelection(freightName);
-                // publishEvent("freightOptionSelected", freightOption.name);
-                // console.log(freightName);
+                this.dispatchEvent(
+                  new CustomEvent("onFreightOptionSelected", {
+                      detail: freightOption,
+                      bubbles: true
```

Comment:
Do we need to bubble this event?

Reply:
im not 100% sure but i dont think so as we have it inside a CustomEvent and this means it should be available to the whole site ??? will need to talk about this !